pkg/prometheus/promcfg.go

```diff
@@ -3512,6 +3505,7 @@ func (cg *ConfigGenerator) generateScrapeConfig(
 			configs[i] = cg.addOAuth2ToYaml(configs[i], s, config.OAuth2)
 			configs[i] = cg.addProxyConfigtoYaml(ctx, configs[i], sc.GetNamespace(), store, config.ProxyConfig)
 			configs[i] = cg.addBasicAuthToYaml(configs[i], s, config.BasicAuth)
+			configs[i] = cg.WithMinimumVersion("2.23.0").addFiltersToYaml(configs[i], config.Filters)
```

Comment:
In fact, Docker SD already requires 2.23.0 (though it's not enforced in `validateDockerSDConfigs` and should be fixed in another PR).
```suggestion
			configs[i] = cg.addFiltersToYaml(configs[i], config.Filters)
```

Reply:
Got it. I will remove the version check here and then make a PR for the Docker refactor and handle it there.